pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -858,15 +858,14 @@ func (p *PriorityQueue) podMatchesEvent(podInfo *framework.QueuedPodInfo, event
 		return true
 	}
 
-	evtToPlugins := p.clusterEventMap[podInfo.Pod.Spec.SchedulerName]
-	if evtToPlugins == nil {
-		return false
-	}
-
-	// Don't directly use evtToPlugins[clusterEvent].Intersection(podInfo.UnschedulablePlugins) as
 	// upon the same resource type, we need to check if the ActionType is *compatible* instead
 	// of *identical*.
-	for evt, nameSet := range evtToPlugins {
+	for evt, nameSet := range p.clusterEventMap {
+		// Although rare, it's still allowed for plugins to return framework.WildCardEvent.
+		if evt == framework.WildCardEvent {
```

Comment:
But we still need to see if the plugins match.

I think a more readable implementation is:

```golang
eventMatch := evt == framework.WildCardEvent || (evt.Resource == clusterEvent.Resoruce && evt.ActionType&clusterEvent.ActionType != 0)
return eventMatch && pluginsIntersect(nameSet, podInfo.UnschedulablePlugins)
```

Note that I won't be giving an LGTM until we have performance evaluations. Doesn't need to be at integration level. I would prefer a performance test that stresses just `MoveAllToActiveOrBackoffQueue`.

Reply:
> But we still need to see if the plugins match.

Thanks for the catch! Updated.

> eventMatch && pluginsIntersect(nameSet, podInfo.UnschedulablePlugins)

We should continue to search if it's false. See this UT:

```go
{
	name:    "pod's failed plugin registers fine-grained event",
	podInfo: newQueuedPodInfoNoTimestamp(st.MakePod().Name("p").Obj(), "bar"),
	event:   NodeTaintChange,
	clusterEventMap: map[framework.ClusterEvent]sets.String{
		framework.ClusterEvent{Resource: framework.Node, ActionType: framework.All}:             sets.NewString("foo"),
		framework.ClusterEvent{Resource: framework.Node, ActionType: framework.UpdateNodeTaint}: sets.NewString("bar"),
	},
	want: true,
},
```

> Note that I won't be giving an LGTM until we have performance evaluations.

That's definitely fair. I have a PR to introduce a churn operator to scheduler perf framework test. With that, we can easily verify the performance.

> I would prefer a performance test that stresses just MoveAllToActiveOrBackoffQueue.

This sounds good to me. I will see if I can add one as a bonus.